### PR TITLE
Remove local forwarder channel

### PR DESCRIPTION
**Pull request by littleaj**
I did a series of modified reverts of commits related to local forwarder functionality.

I kept the extraction of TelemetryChannelBase and paramiterization of TelemetryBuffer, etc. because that seems to be working fine and could be quite difficult to push down any changes that may have been added into the extracted superclass.

### CODE REVIEW

File: azure-application-insights-spring-boot-starter/src/main/java/com/microsoft/applicationinsights/autoconfigure/ApplicationInsightsTelemetryAutoConfiguration.java [hunk 1]
```diff
@@ -178,22 +176,11 @@ public TelemetryClient telemetryClient(TelemetryConfiguration configuration) {
     @Bean
     @ConditionalOnMissingBean
     public TelemetryChannel telemetryChannel(TelemetryConfiguration configuration) {
-        boolean hasLfEndpoint = StringUtils.isNotBlank(environment.getProperty("azure.application-insights.channel.local-forwarder.endpoint-address"));
         boolean hasInprocEndpoint = StringUtils.isNotBlank(environment.getProperty("azure.application-insights.channel.in-process.endpoint-address"));
-        if (hasLfEndpoint && hasInprocEndpoint) {
+        if (hasInprocEndpoint) {
```
Comment (trask) [inline azure-application-insights-spring-boot-starter/src/main/java/com/microsoft/applicationinsights/autoconfigure/ApplicationInsightsTelemetryAutoConfiguration.java]: I think this whole condition goes away

File: azure-application-insights-spring-boot-starter/src/main/java/com/microsoft/applicationinsights/autoconfigure/ApplicationInsightsTelemetryAutoConfiguration.java [hunk 2]
```diff
@@ -178,22 +176,11 @@ public TelemetryClient telemetryClient(TelemetryConfiguration configuration) {
...
             throw new IllegalConfigurationException("SDK cannot have two channels, please either remove Local Forwarder Endpoint, or In Process Endpoint");
         }
```
Comment (trask) [inline azure-application-insights-spring-boot-starter/src/main/java/com/microsoft/applicationinsights/autoconfigure/ApplicationInsightsTelemetryAutoConfiguration.java]: I think this whole condition goes away

Comment (trask) [inline azure-application-insights-spring-boot-starter/src/main/java/com/microsoft/applicationinsights/autoconfigure/ApplicationInsightsTelemetryAutoConfiguration.java]: I think this whole conditional goes away

**Comment by littleaj** [comment]
\azp run

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,9 @@
 # CHANGELOG
 
+# Version 2.6.0-BETA.2
+* Removed Local Forwarder Channel. Local Forwarder is now a deprecated solution.
+The classes and configuration elements which supported this solution have been removed.
```
Comment (trask) [inline CHANGELOG.md]: 👍

**Pull request merged**